superset/assets/src/modules/Registry.js

```diff
@@ -0,0 +1,52 @@
+export default class Registry {
+  constructor(name = '') {
+    this.name = name;
+    this.items = {};
+    this.promises = {};
+  }
+
+  has(key) {
+    const item = this.items[key];
+    return item !== null && item !== undefined;
+  }
+
+  registerValue(key, value) {
+    this.items[key] = { value };
+    delete this.promises[key];
```

Comment:
This may happen a lot though for the presets. For example, registering a preset that contains all common charts then override `bar` chart with a new plugin. Some presets may reference another preset and override certain charts. 

I think throwing an error would be too harsh. 
Can add warning but then that will mean for every plugin we have to add `.remove()` call before `.register()` to avoid the warning. If every plugin does that, would a warning still be useful?

Reply:
Oh that's a fair point, sgtm 👍